### ISSUE TITLE
Add a helper to intercept calls to the standard lib's global logger

### ISSUE THREAD

**Issue by heyimalex** [uber-go/zap#187]
Currently I'm intercepting all calls to the global logger and sending them to zap. Some libraries I use ([facebookgo/grace](https://github.com/facebookgo/grace)) use the std logging functions and don't provide a way to pass in a logger, so I can't use zwrap. Ultimately all of my logs are going into elasticsearch (so, json) and I didn't want to have to add custom filtering to format these bare log lines.

The `log.SetOutput` function takes an `io.Writer` where each line is written with a single `Write` call, so intercepting looks something like this:

```go
func InterceptGlobalLogs(l zap.Logger, printAt zap.Level) {
    // Disable timestamp flags as they're obviated by the timestamp in the
    // entry. Include the file and line.
    log.SetFlags(log.Llongfile)
    log.SetOutput(&logInterceptor{l,printAt})
}

type logInterceptor struct {
    logger  zap.Logger
    printAt zap.level
}

func (i *logInterceptor) Write(line []byte) (int, error) {
    i.logger.Log(i.printAt, line[:len(line)-1])
    return len(line), nil
}
```

My version actually parses out the file and line into fields as well. Obviously we can't stop everything from writing to stdout, and ideally facebookgo/grace should just be fixed to allow passing in a stdlib-like logger, but maybe this would still be useful. Just a thought.

**Comment by akshayjshah** [comment]
I hadn't thought of this, but it's a good idea! I'll ping this issue when we're through making breaking changes to the core APIs (we're trying to make it easier to compose loggers) - if you're still interested, I'd love a PR. If not, someone else can carry your idea forward.

**Comment by heyimalex** [comment]
Also [http.Server's `ErrorLog`](https://golang.org/pkg/net/http/#Server) is typed as a `*log.Logger` so wrapping in an interface-compatible way like zwrap does won't work. Pretty annoying, but exposing interceptor in some way might be a good idea, code is pretty much the same either way.

**Comment by akshayjshah** [comment]
The more I think about this, the more I like it - great suggestion, @heyimalex. We'll ship this as part of the 1.0 release.

**Comment by heyimalex** [comment]
Awesome! Anecdotally it's been very helpful. We've been using it to intercept logs for three apis that only accept `*log.Logger` in addition to the global logger, and it's so much easier than parsing out lines after the fact with logstash.

**Comment by akshayjshah** [comment]
Landed on dev in #271.
